### PR TITLE
Pinkesh/migrate remaining components

### PR DESCRIPTION
**Pull request by pinkeshmars**
### Description
Provide a brief overview of what this documentation update is about. Explain what sections or topics are being added or revised.

Linear ticket and [magic word](https://linear.app/docs/github#link-prs) Fixes DEVR-569  

## Type of change
- [ ] Typo fix 
- [x] New feature 
- [ ] Enhancement to current docs
- [ ] Removed outdated references
- [ ] Update assets

### CODE REVIEW

**Comment by PoojaB26** [comment]
@pinkeshmars Please do one more thing in this PR or whenever you migrate an old topic:
To ensure these links are redirected from old urls: please go through `firebase.json` file, look for the old URL and then replace "/" with the new URL (public URL).

File: docs/resources/ui/components/built-in-components/color-picker.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+slug: color-picker
+title: Color Picker
+tags: [Actions, Components]
```
Comment (PoojaB26) [inline docs/resources/ui/components/built-in-components/color-picker.md:4]: Add the tag for widget category (like Base Elements or Layout Elements) so they can be grouped together in the generated tag page.

File: docs/resources/ui/components/built-in-components/datetime-picker.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+slug: datetime-picker
+title: DateTime Picker
+tags: [Actions, Components]
```
Comment (PoojaB26) [inline docs/resources/ui/components/built-in-components/datetime-picker.md:4]: Same comment as color picker.

File: docs/resources/ui/components/built-in-components/flippablecard.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
+---
+slug: flippable-card
+title: Flippable Card
+tags: [Layout Elements, Components]
```
Comment (PoojaB26) [inline docs/resources/ui/components/built-in-components/flippablecard.md:4]: Ensure this is under Layout Elements in widget builder.

File: docs/resources/ui/widgets/built-in-widgets/list-grid.md [hunk 4]
```diff
@@ -230,7 +232,20 @@ GridView provides a two-dimensional array of children. It is the widget of choic
 
 Like ListView, GridView only renders the visible items, making it efficient for displaying large collections of elements. GridView supports multiple configurations for column count, spacing, aspect ratio, and scroll directions, offering robust customization options for diverse layout needs.
 
+:::info
+GridView and StaggeredView are almost similar widgets in FlutterFlow, with the primary difference being that StaggeredView is typically used to create a layout similar to the Pinterest app and has fewer properties than GridView.
```
Comment (PoojaB26) [inline docs/resources/ui/widgets/built-in-widgets/list-grid.md:236]: Pinterest example should be added as an example at the end and not as part of the description as people may not know Pinterest app that well. You should talk about how StaggeredView is different from Grid View in terms of fixed vs unfixed size of children, and any other major differences. And then add an additional point about the Pinterest app example.

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md [hunk 5]
```diff
@@ -0,0 +1,527 @@
+---
+slug: /resources/forms/checkbox
+title: Checkboxes
```
Comment (PoojaB26) [inline docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:3]: Just Checkbox

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+tags: [Widgets, Form Elements]
+keywords: [FlutterFlow, Checkbox, CheckboxGroup, CheckboxListTile, Multiple Choice]
+description: Learn how to add Checkbox, CheckboxGroup, and CheckboxListTile widget in your FlutterFlow app.
+sidebar_position: 3
+---
+# Checkboxes
```
Comment (PoojaB26) [inline docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:9]: Just Checkbox

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md [hunk 7]
```diff
@@ -0,0 +1,527 @@
...
+In FlutterFlow, checkboxes are versatile input widgets used to capture binary choices from users, such as true/false or yes/no options. They are ideal for situations where you need to present users with options that can be individually selected or deselected. FlutterFlow provides three primary checkbox widgets: [**Checkbox**](#checkbox), [**CheckboxListTile**](#checkboxlisttile), and [**CheckboxGroup**](#CheckboxGroup). Each of these widgets offers distinct features and use cases, making it easy to tailor your app's interface to your specific needs.
```
Comment (PoojaB26) [inline docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:10]:
```suggestion
In FlutterFlow, a checkbox is a versatile input widget used to capture binary choices from users, such as true/false or yes/no options. It is ideal for situations where you need to present users with options that can be individually selected or deselected. FlutterFlow provides three primary variations of the checkbox widget: [**Checkbox**](#checkbox), [**CheckboxListTile**](#checkboxlisttile), and [**CheckboxGroup**](#CheckboxGroup). Each of these widgets offers distinct features and use cases, making it easy to tailor your app's interface to your specific needs.
```

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md (hunk 6 above)
Comment (PoojaB26) [reply docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:9]: To differentiate just say Checkbox Widgets here on the title and then Checkbox later as h2

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+The **CheckboxListTile** widget combines the functionality of a checkbox with a [ListTile](../../../../ui/widgets/built-in-widgets/list-grid.md#listtile-widget), providing a more comprehensive option for displaying checkboxes alongside additional information. Unlike the Checkbox this widget includes a title, and an optional subtitle, all within a single, cohesive element.
+
+CheckboxListTile is ideal for use cases where you want to provide more context or descriptive text alongside the checkbox, such as in a settings menu or a form with detailed options.
+
+
+## CheckboxGroup
+
+The **CheckboxGroup** widget allows you to present a group of checkboxes as a single entity. This is particularly useful when you want users to select multiple options from a list. Each checkbox within the group can be checked or unchecked independently of the others.
+
+:::tip[Widget State]
```
Comment (PoojaB26) [inline docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:271]: This should be added to the top of page as all the widgets here have a widget state. Add them before talking about the variations in details.

File: docs/resources/control-flow/user-interactivity/forms/forms.md [hunk 9]
```diff
@@ -12,5 +12,5 @@ sidebar_position: 0
 Forms are a fundamental part of many applications, serving as the primary method for users to input and submit data. Whether you're building a simple contact form or a complex multi-step survey, FlutterFlow provides a comprehensive set of tools to create, validate, and manage forms effectively.
 
 :::tip
-In this section, you'll learn how to add form widgets such as [**TextField**](form-widgets/text-field.md), [**Dropdown**](form-widgets/dropdown.md), [**RadioButton**](form-widgets/radiobutton.md), [**CheckboxGroup**](form-widgets/checkbox-group.md) and add [**Validations**](form-validation.md) and [**set**](form-actions/set-form-field.md)/[**reset**](form-actions/reset-form-field.md) actions on these widgets.
+In this section, you'll learn how to add form widgets such as [**TextField**](form-widgets/text-field.md), [**Dropdown**](form-widgets/dropdown.md), [**RadioButton**](form-widgets/radiobutton.md), [**CheckboxGroup**](form-widgets/checkbox.md) and add [**Validations**](form-validation.md) and [**set**](form-actions/set-form-field.md)/[**reset**](form-actions/reset-form-field.md) actions on these widgets.
```
Comment (PoojaB26) [inline docs/resources/control-flow/user-interactivity/forms/forms.md:15]:
```suggestion
In this section, you'll learn how to add form widgets such as [**TextField**](form-widgets/text-field.md), [**Dropdown**](form-widgets/dropdown.md), [**RadioButton**](form-widgets/radiobutton.md), [**Checkbox Widgets**](form-widgets/checkbox.md) and add [**Validations**](form-validation.md) and [**set**](form-actions/set-form-field.md)/[**reset**](form-actions/reset-form-field.md) actions on these widgets.
```

**Review by PoojaB26 — CHANGES_REQUESTED** [review]
Did a quick review of just the text! Will do another review once you complete the PR.

File: docs/resources/ui/components/built-in-components/color-picker.md (hunk 1 above)
Comment (pinkeshmars) [reply docs/resources/ui/components/built-in-components/color-picker.md:4]: Color picker is available as an action only, so added 'Actions' tag. It doesn't fall under any widget category by default.

File: docs/resources/ui/components/built-in-components/datetime-picker.md (hunk 2 above)
Comment (pinkeshmars) [reply docs/resources/ui/components/built-in-components/datetime-picker.md:4]: Again, this is available as action only.

File: docs/resources/ui/components/built-in-components/flippablecard.md (hunk 3 above)
Comment (pinkeshmars) [reply docs/resources/ui/components/built-in-components/flippablecard.md:4]: Yes, it is.
Screenshot 2024-08-06 at 4 47 27 PM

File: docs/resources/ui/widgets/built-in-widgets/list-grid.md (hunk 4 above)
Comment (pinkeshmars) [reply docs/resources/ui/widgets/built-in-widgets/list-grid.md:236]: Good point! Updated.

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md (hunk 5 above)
Comment (pinkeshmars) [reply docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:3]: Done!

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md (hunk 6 above)
Comment (pinkeshmars) [reply docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:8]: Done!

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md (hunk 7 above)
Comment (pinkeshmars) [reply docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:9]: Updated!

File: docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md (hunk 8 above)
Comment (pinkeshmars) [reply docs/resources/control-flow/user-interactivity/forms/form-widgets/checkbox.md:270]: Thanks for catching!

**Comment by pinkeshmars** [comment]
Hey @PoojaB26 This PR is complete (except chat + group chat) and the review comments given so far are addressed.

**Review by PoojaB26 — APPROVED** [review]

**Comment by PoojaB26** [comment]
@pinkeshmars One big change I did in this PR is create a new file under Functions/Actions called Utility Actions and add both color picker and date picker there. In the future, we may wanna add things like "copy to clipboard" or other non-UI utlity actions there as well.

**Pull request merged**